tests/phpunit/tests/media.php

```diff
@@ -3278,6 +3278,78 @@ public function test_wp_get_attachment_image_loading_opt_out_individual() {
 		$this->assertStringNotContainsString( ' loading=', $img );
 	}
 
+	/**
+	 * @ticket 57086
+	 *
+	 * @dataProvider data_wp_get_attachment_image_decoding_attr
+	 *
+	 * @covers ::wp_get_attachment_image
+	 */
+	public function test_wp_get_attachment_image_decoding_attr( $decoding, $expected ) {
+		if ( 'no value' === $decoding ) {
+			$image = wp_get_attachment_image( self::$large_id, 'thumbnail', false, array() );
+		} else {
+			$image = wp_get_attachment_image( self::$large_id, 'thumbnail', false, array( 'decoding' => $decoding ) );
+		}
+
+		$this->assertSame( $expected, $image );
+	}
+
+	/**
+	 * Data provider for test_wp_get_attachment_image_decoding_attr().
+	 *
+	 * @return array[]
+	 */
+	public function data_wp_get_attachment_image_decoding_attr() {
+		return array(
+			'default'     => array(
+				'decoding' => 'no value',
+				'expected' => '<img width="150" height="150" src="http://example.org/wp-content/uploads/2022/11/test-image-large-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="" decoding="async" loading="lazy" />',
```

Comment:
```suggestion
				'expected' => '<img width="150" height="150" src="http://' . WP_TESTS_DOMAIN . '/wp-content/uploads/2022/11/test-image-large-150x150.jpg" class="attachment-thumbnail size-thumbnail" alt="" decoding="async" loading="lazy" />',
```

Further to @felixarntz's comment, in the current format the tests need to change to this to account for local runs. But I'd probably go with Felix's approach of using a different approach to validate the attribute.

Reply:
Updated in https://github.com/WordPress/wordpress-develop/pull/3633/commits/10fcec3d6860702e1e5e77cf9b10f3cf431ab0b5.